src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -229,10 +220,14 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
             {
                 return BindParameterFromRouteValueOrQueryString(parameter, parameter.Name, factoryContext);
             }
-            else
+            else if (parameter.ParameterType.IsInterface)
             {
                 return Expression.Call(GetRequiredServiceMethod.MakeGenericMethod(parameter.ParameterType), RequestServicesExpr);
             }
+            else
+            {
+                return BindParameterFromBody(parameter.ParameterType, allowEmpty: false, factoryContext);
```

Comment:
Yeah, I think 👍🏽 @davidfowl 's comment - could we rely on the presence of a default value to determine if this is optional? e.g.

```C#
MyModel Echo(MyModel model) => model; // AllowEmpty = false

MyModel Echo(MyModel model = null) => model; // AllowEmpty = true
```

If we were absolutely bonkers like MVC, we could also rely on nullability of the parameter to determine optionality:

```C#
MyModel Echo(MyModel? model) => model; // AllowEmpty = true
```

Reply:
Yep. I'm a huge fain of using nullability to determine `AllowEmpty` behavior. I think we should also use it when the `[FromBody]` attribute is applied without an explicit parameter.

Reply:
Behold ye what needs to be done for nullability: https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.DataAnnotations/src/DataAnnotationsMetadataProvider.cs#L465-L568. It would be nice to put this in a shared space.

Btw, it would be a lot easier to infer nullability once we have a source generator.